src/OnlineRetailPortal.Contracts/Models/Product.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OnlineRetailPortal.Contracts
+{
+    public class Product
+    {
+        public string Id { get; set; }
+        public string Name { get; set; }
+        public Price Price { get; set; }
+        public Category Category { get; set; }
+        public Image HeroImage { get; set; }
+        public Address PickupAddress { get; set; }
+        public List<Image> Images { get; set; }
+        public string UserId { get; set; }
+        public DateTime PostDateTime { get; set; }
+        public DateTime PurchasedDate { get; set; }
+        public DateTime ExpirationDate { get; set; }
```

Comment:
How are we going to handle expiration. Instead we can have IsActive flag in Ad/Product

Reply:
product has a property status, so we can mark product active, sold or disabled.